app/middleware/activity.js

```diff
@@ -0,0 +1,47 @@
+const User = require('../models/User')
+const redis = require('../../config/redis')
+const activity = async (req, res, next) => {
+  var redisClient = redis.redisClient
+  var route = req.originalUrl.replace(/\?.*$/, '')
+  if (route === '/user/logout') {
+    var userID = req.user.id.toString()
+    var activityData = await redisClient.lrange(userID, 0, -1)
+    const data = await User.findOne({
+      _id: userID
+    })
+    var activityElement = {
+      route: "",
+      method: "",
+      collectionType: "",
+      id: "",
+    }
+    for (let index = 0; index < activityData.length; index++) {
+      var activityDataElement = activityData[index].split(',')
+      activityElement.route = activityDataElement[0]
+      activityElement.method = activityDataElement[1]
+      activityElement.collectionType = activityDataElement[2]
+      activityElement.id = activityDataElement[3]
+      data.activity.push(activityElement)
+    }
+    await data.update()
+    console.log("DATA")
+    console.log(data)
+    // clear data from redis
+    await redisClient.del(userID)
+  } else if (method != 'GET') {
```

Comment:
> @kmehant Awesome!!
> Also I think we need to add this middleware in every routes except GET requests. In addition to that do you think we need to track user login and logout??

@Rupeshiya I have written check condition here so we can add it everywhere safely 😁